models/flow_evm_tx.go

```diff
@@ -0,0 +1,23 @@
+package models
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type FlowEVMTxData interface {
```

Comment:
I would change naming and placement of all this transactions/calls. 

My suggestion is to name the interface something like: `Transaction` and then have the `DirectCallTx` just be `DirectCall` and `gethTypes.Transaction` can just be `TransactionCall`. Also I would move all of those in the `transaction.go` file so we have all relevant stuff in once place (but I would move decode receipt out into a `receipt.go`)